site/docs/getting-started/tutorials/postgresql_cdc_to_snowflake.md

```diff
@@ -0,0 +1,449 @@
+
+:::note This file defines the schema via a create table statement, but the actual table creation happens in the `init.sql` file, this is just a quirk of the [Datagen](https://github.com/MaterializeInc/datagen) data generator tool.
+:::
+
+```sql title="products.sql"
+CREATE TABLE "public"."products" (
+  "id" int PRIMARY KEY,
+  "name" varchar COMMENT 'faker.internet.userName()',
+  "merchant_id" int NOT NULL COMMENT 'faker.datatype.number()',
+  "price" int COMMENT 'faker.datatype.number()',
+  "status" varchar COMMENT 'faker.datatype.boolean()',
+  "created_at" timestamp DEFAULT (now())
+);
+```
+
+If you take a closer look at the schema definition, you can see that in the `COMMENT` attribute we define Python snippets which actually tell Datagen how to generate fake data for those fields!
+
+Finally, create the `init.sql` file, which contains the database-level requirements to enable Flow to stream CDC data.
+
+```sql title="init.sql"
+CREATE USER flow_capture WITH PASSWORD 'secret' REPLICATION;
+
+GRANT pg_read_all_data TO flow_capture;
+
+CREATE TABLE products (
+  "id" int PRIMARY KEY,
+  "name" varchar COMMENT 'faker.internet.userName()',
+  "merchant_id" int NOT NULL COMMENT 'faker.datatype.number()',
+  "price" int COMMENT 'faker.datatype.number()',
+  "status" varchar COMMENT 'faker.datatype.boolean()',
+  "created_at" timestamp DEFAULT (now())
+);
+
+CREATE TABLE IF NOT EXISTS public.flow_watermarks (slot TEXT PRIMARY KEY, watermark TEXT);
+GRANT ALL PRIVILEGES ON TABLE public.flow_watermarks TO flow_capture;
+CREATE PUBLICATION flow_publication;
+ALTER PUBLICATION flow_publication SET (publish_via_partition_root = true);
+ALTER PUBLICATION flow_publication ADD TABLE public.flow_watermarks, public.products;
+```
+
+In the `init.sql` file, you create the products table and all the database objects Flow requires for real-time CDC streaming.
+
+
+### Configuring PostgreSQL for CDC<a id="configuring-postgresql-for-cdc"></a>
+
+To enable CDC replication in PostgreSQL, several database objects need to be created and configured. These objects facilitate the capture and propagation of data changes to downstream systems. Let's examine each object and its significance in the context of CDC replication:
+
+```sql
+CREATE USER flow_capture WITH PASSWORD 'secret' REPLICATION;
+```
+
+This user is dedicated to the CDC replication process. It is granted the necessary permissions to read all data from the database, allowing it to capture changes across tables efficiently. In a production environment, make sure you use a more secure password than what is in the example.
+
+```sql
+GRANT pg_read_all_data TO flow_capture;
+```
+
+Granting the `pg_read_all_data` privilege to the `flow_capture` user ensures that it can access and read data from all tables in the database, essential for capturing changes.
```

Comment:
It might be useful to mention here that `pg_read_all_data` is used for convenience in the tutorial but is not a hard requirement, since it is possible to grant a more restricted set of permissions and our docs discuss that in more detail.

Reply:
Thanks, that's a good flag!